src/org/labkey/test/components/html/BootstrapMenu.java

```diff
@@ -196,26 +196,40 @@ public static Locator.XPathLocator dropdownMenu()
             return dropdownMenu(dropdownToggle());
         }
 
-        private static Locator.XPathLocator dropdownMenu(Locator.XPathLocator toggleLoc)
+        private static Locator.XPathLocator dropdownMenu(Locator.XPathLocator toggleLoc, boolean isDropup)
```

Comment:
Note: you can union XPathLocators if you want so you can avoid the need to pass in the isDropup param

`Locator.XPathLocator.union(Locator.tagWithClass("*", "dropup"), Locator.tagWithClass("*", "dropdown")).withChild(childLocators)`

Reply:
Thanks! I reverted most of the changes and replaced them with your suggestion. Going to keep that in my grab bag of tricks.